cloud/gcp/templates/gke_autopilot/main.tf

```diff
@@ -0,0 +1,187 @@
+  name     = var.cluster_name
+  location = var.region
+}
+
+provider "kubernetes" {
+  host                   = "https://${data.google_container_cluster.civiform_cluster.endpoint}"
+  token                  = data.google_client_config.default.access_token
+
+  // protect traffic to the KCP with TLS
+  cluster_ca_certificate = base64decode(
+    data.google_container_cluster.civiform_cluster.master_auth[0].cluster_ca_certificate,
+  )
+
+  // The access token is stored in TF state but expires hourly, this plugin refreshes it on each Terraform apply
+  exec {
+    api_version = "client.authentication.k8s.io/v1beta1"
+    command     = "gke-gcloud-auth-plugin"
+  }
+}
+
+locals {
+  server_port_name       = "http-server"
+  services_ip_range_name = "services-range"
+  pods_ip_range_name     = "pods-range"
+  fully_qualified_service_account = "${var.cluster_service_account_name}@${var.project_id}.iam.gserviceaccount.com"
+}
+
+resource "kubernetes_deployment_v1" "civiform_server" {
+  metadata {
+    name = "civiform-server-deployment"
+  }
+
+  spec {
+    selector {
+      match_labels = {
+        app = "civiform-server"
+      }
+    }
+
+    template {
+      metadata {
+        labels = {
+          app = "civiform-server"
+        }
+      }
+
+      spec {
+        container {
+          name  = "civiform-server-container"
+          image = var.server_image
+
+          port {
+            container_port = var.port
+            name           = local.server_port_name
+          }
+
+          security_context {
+            allow_privilege_escalation = false
+            privileged                 = false
+            read_only_root_filesystem  = false
+          }
+
+          liveness_probe {
+            http_get {
+              path = "/playIndex"
+              port = var.port
+            }
+
+            initial_delay_seconds = 60
+            period_seconds        = 10
+          }
+
+          env {
+            name  = "PORT"
+            value = var.port
+          }
+          env {
+            name = "CIVIFORM_APPLICANT_IDP"
+            value = "disabled"
+          }
+          env {
+            name = "APPLICANT_OIDC_CLIENT_ID"
+            value = "client"
+          }
+          env {
+            name = "APPLICANT_OIDC_CLIENT_SECRET"
+            value = "notarealsecret"
+          }
+          env {
+            name = "APPLICANT_OIDC_DISCOVERY_URI"
+            value = "https://civiform-staging.us.auth0.com/.well-known/openid-configuration"
+          }
+          env {
+            name = "CIVIFORM_APPLICANT_IDP"
+            value = "auth0"
+          }
+          env {
+            name = "WHITELABEL_CIVIC_ENTITY_SHORT_NAME"
+            value = "CF on GCP"
+          }
+          env {
+            name = "WHITELABEL_CIVIC_ENTITY_FULL_NAME"
+            value = "CiviForm on GCP"
+          }
+          env {
+            name = "SECRET_KEY"
+            value = "K5SgucxBYC3xJwNcGWZV1Y7uASmrc"
```

Comment:
**suggestion:** Can we put a fake value in here or use a gcp secrets manager?

Reply:
Replaced with an obviously insecure value